examples/glue_benchmark/glue_mrpc.py

```diff
@@ -0,0 +1,193 @@
+    "preset",
+    None,
+    "The model preset(eg. For bert it is 'bert_base_en', 'bert_tiny_en_uncased')",
+)
+flags.DEFINE_float(
+    "learning_rate", 0.005, "The learning_rate for the optimizer."
+)
+flags.DEFINE_integer("epochs", 2, "No of Epochs.")
+flags.DEFINE_integer("batch_size", 8, "Batch Size.")
+
+
+FLAGS = flags.FLAGS
+
+
+def load_data():
+    """
+    Load GLUE/MRPC dataset, and convert the dictionary format to (features, label),
+    where features is a tuple of all input sentences.
+    """
+
+    feature_names = ("sentence1", "sentence2")
+
+    def split_features(x):
+        # GLUE comes with dictonary data, we convert it to a uniform format
+        # (features, label), where features is a tuple consisting of all
+        # features.
+        features = tuple([x[name] for name in feature_names])
+        label = x["label"]
+        return (features, label)
+
+    train_ds, test_ds, validation_ds = tfds.load(
+        "glue/mrpc",
+        split=["train", "test", "validation"],
+    )
+
+    train_ds = train_ds.map(split_features, num_parallel_calls=tf.data.AUTOTUNE)
+    test_ds = test_ds.map(split_features, num_parallel_calls=tf.data.AUTOTUNE)
+    validation_ds = validation_ds.map(
+        split_features, num_parallel_calls=tf.data.AUTOTUNE
+    )
+    return train_ds, test_ds, validation_ds
+
+
+def load_model_and_preprocessor(model, preset):
+    for name, symbol in keras_nlp.models.__dict__.items():
+        if inspect.isclass(symbol) and issubclass(symbol, keras.Model):
+            if model and name != model:
+                continue
+            if not hasattr(symbol, "from_preset"):
+                continue
+            for _preset in symbol.presets:
+                if preset and _preset != preset:
+                    continue
+                model = symbol.from_preset(preset)
+                print(f"Using model {name} with preset {preset}")
+                preprocessor = keras_nlp.models.__dict__[
+                    name.replace("Backbone", "Preprocessor")
+                ].from_preset(preset)
+
+                return model, preprocessor
+
+    raise ValueError(f"Model {model} or preset {preset} not found.")
+
+
+def create_model(model, preset):
+    # output_dim
+    output_dim = 2
+    # select backbone
+
+    if "backbone" not in FLAGS.model.lower():
+        backbones = [
+            i
+            for i in keras_nlp.models.__dict__
+            if "Backbone" in i
+            and i not in ["OPTBackbone", "GPT2Backbone", "WhisperBackbone"]
+        ]
+        raise ValueError(
+            f"Please enter a Backbone, from this list - {backbones}, the script will take care of the downstream layers."
+        )
+
+    # Build the model
+    backbone, preprocessor = load_model_and_preprocessor(
+        model=model, preset=preset
+    )
+    # Freeze
+    if FLAGS.freeze_backbone:
+        backbone.trainable = False
+    else:
+        backbone.trainable = True
+    # If the model has pooled_output
```

Comment:
I would avoid this special casing and actually use the `XXClassifier` classes. This will be important for things like BART, which is going to look substantially different (feed the input sequence twice to the encoder and decoder and use the last token representation in the decoder block).

We could also have this script by default use the compilation defaults in the `XXClassifier` classes, which would be a great way to be able to test them out! And make sure they are reasonable defaults.

Reply:
thanks again for your comments, in the next commit I will change it to `XXClassifier` .